src/cloudant/index.py

```diff
@@ -262,3 +267,201 @@ def delete(self):
+            Fields containing other data (numbers, objects, arrays)
+            can not be used.
+        :param int group_limit: Maximum group count.
+            This field can only be used if ``group_field`` query parameter
+            is specified.
+        :param group_sort: A JSON field that defines the order of the
+            groups in a search using ``group_field``. The default sort order
+            is relevance. This field can have the same values as the sort field,
+            so single fields as well as arrays of fields are supported.
+        :param int limit: Limit the number of the returned documents to the
+            specified count. In case of a grouped search, this parameter limits
+            the number of documents per group.
+        :param query: A Lucene query in the form of ``name:value``.
+            If name is omitted, the special value ``default`` is used.
+        :param ranges: A JSON facet syntax that reuses the standard Lucene
+            syntax to return counts of results which fit into each specified
+            category. Inclusive range queries are denoted by brackets.
+            Exclusive range queries are denoted by curly brackets.
+            For example ``ranges={"price":{"cheap":"[0 TO 100]"}}`` has an
+            inclusive range of 0 to 100.
+            Requires the index to have faceting enabled.
+        :param sort: A JSON string of the form ``fieldname<type>`` for ascending
+            or ``-fieldname<type>`` for descending sort order. Fieldname is the
+            name of a string or number field and type is either number or string
+            or a JSON array of such strings. The type part is optional and
+            defaults to number.
+        :param str stale: Allow the results from a stale index to be used. This
+            makes the request return immediately, even if the index has not been
+            completely built yet.
+        :param list highlight_fields: A list of fields which should be
+            highlighted.
+        :param str highlight_pre_tag: A string inserted before the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param str highlight_post_tag: A string inserted after the highlighted
+            word in the highlights output.  Defaults to ``<em>``.
+        :param int highlight_number: A number of fragments returned in
+            highlights. If the search term occurs less often than the number of
+            fragments specified, longer fragments are returned.  Default is 1.
+        :param int highlight_size: A number of characters in each fragment for
+            highlights.  Defaults to 100 characters.
+        :param list include_fields: A list of field names to include in search
+            results. Any fields included must have been indexed with the
+            ``store:true`` option.
+
+        :returns: Search query result data in JSON format
```

Comment:
Yes, I used the content from Cloudant documentation and made some changes to follow the docstring design in the Query and View callable function.